public/app/features/plugins/plugin_loader.ts

```diff
@@ -42,6 +42,13 @@ grafanaUI.DataSourcePlugin = grafanaData.DataSourcePlugin;
 grafanaUI.AppPlugin = grafanaData.AppPlugin;
 grafanaUI.DataSourceApi = grafanaData.DataSourceApi;
 
+// Patch in an old function to stop old plugins from crashing
+if (!(grafanaData as any).getColorForTheme) {
+  (grafanaData as any).getColorForTheme = (color: string, theme: grafanaData.GrafanaTheme) => {
+    return theme.visualization.getColorByName(color);
+  };
+}
+
```

Comment:
Yeah, this is a tough one. 

I would probably lean towards not keeping the shim, but try to help updating the dependent plugins to use `getColorByName()`. My reasoning would be that `getColorForTheme()` was deprecated, and [cleaning it up](https://github.com/grafana/grafana/pull/49519) in this major version (v9) was probably the best we could do. 

On the other hand if we would like to revert this breaking change, maybe we should rather put the shim in `@grafana/data` and export it with a note?

Reply:
> On the other hand if we would like to revert this breaking change, maybe we should rather put the shim in @grafana/data and export it with a note?

Not sure how to do that, without making part of the exported typing you mean? 

> I would probably lean towards not keeping the shim, but try to help updating the dependent plugins to use getColorByName(). My reasoning would be that getColorForTheme() was deprecated, and https://github.com/grafana/grafana/pull/49519 in this major version (v9) was probably the best we could do.

@leventebalogh can you fix this in the singlestat panel and deploy a new version? The maybe we won't need this